### ISSUE TITLE
Text box state changes in page editor

### ISSUE THREAD

**Issue by aerwin** [maqetta/maqetta#2459]
Chrome/Mac...

Working through claro theme editor test plan, I have a theme with changes made to buttons and text boxes. In addition, I've made the Hover state of both button and text box to have a bigger font size.

When just viewing in the page editor, things look fine:

![screenshot](http://i.imgur.com/FjxYI.png)

But, if I mouse over the text box in the theme editor, the text box gets rendered with what is set in the Hover state in the edited theme:

![screenshot](http://i.imgur.com/W1sAd.png)

**Comment by jhpedemonte** [comment]
@billreed63, push to M8?

@JonFerraiolo

**Comment by JonFerraiolo** [comment]
@billreed63 - isn't this fixed? I remember we fixed a bug with the geometry for the transparent overlays (DIV.enableWidgetFocusFrame) recently.

**Comment by billreed63** [comment]
yup it is fixed

**Comment by aerwin** [comment]
Reopening... seeing in FF on Mac right now:

Normal rendering:

![GitHub Logo](http://i.imgur.com/b0vrg.png)

Rendering when hover over text box:

![GitHub Logo](http://i.imgur.com/SWDSR.png)

**Comment by JonFerraiolo** [comment]
Is this the same issue as #2456?

**Comment by aerwin** [comment]
One's in theme editor and one's in page editor, so I don't know.

**Comment by billreed63** [comment]
This looks like the same issue as #2456. I can't reproduce it on FF but I can make the hover happen sometimes on chrome I have to move the mouse very slowly from below the text box to over the text box. Some times I get the hover state of the text box. If I stop moving the mouse as soon as I get the hover state it sticks, but if I move the mouse (still over the textbox) the hover state is removed...

**Comment by JonFerraiolo** [comment]
> One's in theme editor and one's in page editor, so I don't know.

Right. Definitely leave this issue open separately. After thinking about things some more, I expect there are different code paths for theme editor overlays than page editor overlays.

**Comment by jhpedemonte** [comment]
-> M8

**Comment by JonFerraiolo** [comment]
@peller 

OK, I studied this issue some. 

We had the same problem with M6 (and probably forever)

What's going on is that Dijit is listening to the mouseover event and immediately adding a dijitHover class to the root node, and this is happening before Maqetta can intercept the mouseover event and an appropriate editFeedback rectangle over the widget. This seems wrong. Hover processing should only occur after a timing delay, it shouldn't happen immediately with the mouseover.

The best fix that I can think of would be to have Maqetta's mouseover handler set useCapture=1 so that it would be sure to get the mouseover event before the widget did, and then have it call stopEvent() to prevent dijit from getting the mouseover.

-> me for M8 (@JonFerraiolo)
